File: model/parse.go
package model

// StayOutList 외박 신청 내역 파싱하여 저장할 구조체
type StayOutList struct {
	OutStayFrDt  []string
	OutStayToDt  []string
	OutStayStGbn []string
}

// PointList 상벌점 내역 파싱하여 저장할 구조체
type PointList struct {
	CmpScr        []string
	LifSstArdGbn  []string
	ArdInptDt     []string
	LifSstArdCtnt []string
}

// RequestInfo request 보내기 위한 년, 학기, 학번, 학생 이름 저장할 구조체
type RequestInfo struct {
	YY       string
	TmGbn    string
	SchregNo string
	StdKorNm string
}

File: src/model/parse.ts
// 디렉토리 구조를 위한 임시 파일입니다.
interface StayOutList {
  OutStayFrDt: string[];
  OutStayToDt: string[];
  OutStayStGbn: string[];
}

// PointList Structure to parse and store reward/punishment details
interface PointList {
  CmpScr: string[];
  LifSstArdGbn: string[];
  ArdInptDt: string[];
  LifSstArdCtnt: string[];
}

// Structure to store year, semester, student number, and student name for sending RequestInfo request
interface RequestInfo {
  YY: string;
  TmGbn: string;
  SchregNo: string;
  StdKorNm: string;
}
